### PR TITLE
[Feature][Connector V2] expose configurable options in JDBC

### PR DESCRIPTION
**Pull request by john8628**
## Purpose of this pull request
https://github.com/apache/incubator-seatunnel/issues/3385

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by ashulin — DISMISSED** [review]

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java [hunk 1]
```diff
@@ -25,74 +27,74 @@
...
 public class JdbcConfig implements Serializable {
 
-    public static final String URL = "url";
+    public static final Option<String> URL = Options.key("url").stringType().noDefaultValue().withDescription("url");
 
-    public static final String DRIVER = "driver";
+    public static final Option<String> DRIVER =  Options.key("driver").stringType().noDefaultValue().withDescription("driver");
 
-    public static final String CONNECTION_CHECK_TIMEOUT_SEC = "connection_check_timeout_sec";
+    public static final Option<Integer> CONNECTION_CHECK_TIMEOUT_SEC = Options.key("connection_check_timeout_sec").intType().noDefaultValue().withDescription("connection check time second");
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java]: In the doc, connection_check_timeout_sec have default value : 30
image

Comment (john8628) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java]: OK,done

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java]: image
Modify this value too, use `CONNECTION_CHECK_TIMEOUT_SEC.defaultValue()` replaced.

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java [hunk 2]
```diff
@@ -25,74 +27,76 @@
...
 
     //source config
-    public static final String PARTITION_COLUMN = "partition_column";
-    public static final String PARTITION_UPPER_BOUND = "partition_upper_bound";
-    public static final String PARTITION_LOWER_BOUND = "partition_lower_bound";
-    public static final String PARTITION_NUM = "partition_num";
+    public static final Option<String> PARTITION_COLUMN = Options.key("partition_column").stringType().noDefaultValue().withDescription("partition column");
+    public static final Option<String> PARTITION_UPPER_BOUND = Options.key("partition_upper_bound").stringType().noDefaultValue().withDescription("partition upper bound");
+    public static final Option<String> PARTITION_LOWER_BOUND = Options.key("partition_lower_bound").stringType().noDefaultValue().withDescription("partition lower bound");
+    public static final Option<String> PARTITION_NUM = Options.key("partition_num").stringType().noDefaultValue().withDescription("partition num");
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcConfig.java:72]:
```suggestion
    public static final Option<String> PARTITION_UPPER_BOUND = Options.key("partition_upper_bound").longType().noDefaultValue().withDescription("partition upper bound");
    public static final Option<String> PARTITION_LOWER_BOUND = Options.key("partition_lower_bound").longType().noDefaultValue().withDescription("partition lower bound");
    public static final Option<String> PARTITION_NUM = Options.key("partition_num").intType().noDefaultValue().withDescription("partition num");
```

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/options/JdbcConnectionOptions.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
     implements Serializable {
     private static final long serialVersionUID = 1L;
 
-    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = 30;
+    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = org.apache.seatunnel.connectors.seatunnel.jdbc.config.JdbcConfig.CONNECTION_CHECK_TIMEOUT_SEC.defaultValue();
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/options/JdbcConnectionOptions.java]:
```suggestion
    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = JdbcConfig.CONNECTION_CHECK_TIMEOUT_SEC.defaultValue();
```

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/options/JdbcConnectorOptions.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
     implements Serializable {
     private static final long serialVersionUID = 1L;
 
-    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = 30;
+    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = org.apache.seatunnel.connectors.seatunnel.jdbc.config.JdbcConfig.CONNECTION_CHECK_TIMEOUT_SEC.defaultValue();
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/options/JdbcConnectorOptions.java]:
```suggestion
    private static final int DEFAULT_CONNECTION_CHECK_TIMEOUT_SEC = JdbcConfig.CONNECTION_CHECK_TIMEOUT_SEC.defaultValue();
```

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/sink/JdbcSinkFactory.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public String factoryIdentifier() {
+        return "Jdbc";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder().required(URL, DRIVER, QUERY).optional(USER, PASSWORD, MAX_RETRIES, CONNECTION_CHECK_TIMEOUT_SEC, BATCH_SIZE,
+                BATCH_INTERVAL_MS, IS_EXACTLY_ONCE, XA_DATA_SOURCE_CLASS_NAME, MAX_COMMIT_ATTEMPTS, TRANSACTION_TIMEOUT_SEC, TABLE, PRIMARY_KEYS)
+                .build();
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/sink/JdbcSinkFactory.java]:
```suggestion
        return OptionRule.builder().required(URL, DRIVER).exclusive(QUERY, TABLE).optional(USER, PASSWORD, MAX_RETRIES, CONNECTION_CHECK_TIMEOUT_SEC, BATCH_SIZE,
                BATCH_INTERVAL_MS, IS_EXACTLY_ONCE, XA_DATA_SOURCE_CLASS_NAME, MAX_COMMIT_ATTEMPTS, TRANSACTION_TIMEOUT_SEC, PRIMARY_KEYS)
                .build();
```

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by ashulin — APPROVED** [review]
+1

**Pull request merged**